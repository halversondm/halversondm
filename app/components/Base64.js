/**
 * Created by Daniel on 6/26/2016.
 */
"use strict";

import React from "react";

const Base64 = React.createClass({
  getInitialState() {
    return {
      encodeInput: "",
      encodeOutput: "",
      decodeInput: "",
      decodeOutput: ""
    };
  },
  setEncodeInput(event) {
    this.setState({encodeInput: event.target.value});
  },
  goEncode(event) {
    const encodeOutput = window.btoa(this.state.encodeInput);
    this.setState({encodeOutput: encodeOutput});
  },
  clearEncode(event) {
    this.setState({encodeInput: "", encodeOutput: ""});
  },
  setDecodeInput(event) {
    this.setState({decodeInput: event.target.value});
  },
  goDecode(event) {
    const decodeOutput = window.atob(this.state.decodeInput);
    this.setState({decodeOutput: decodeOutput});
  },
  clearDecode(event) {
    this.setState({decodeInput: "", decodeOutput: ""});
  },
  render() {
    return <div>
      <h4 className="text-success">An all Bootstrap and React implementation of
        converting your data to and from Base64 encoding. No tracking of
        your data is done. No calls to other services. It's all in your
        browser.</h4>
      <tabset>
        <tab heading="Encode" id="encodeTab">
          <form className="form-horizontal" role="form">
            <h2>Encode Data to Base64</h2>
            <div className="form-group">
              <label forHtml="encodeInput" className="col-sm-2 control-label">Un-encoded</label>
              <div className="col-sm-10">
                <textarea className="form-control" id="encodeInput"
                          value={this.state.encodeInput}
                          onChange={this.setEncodeInput} required></textarea>
              </div>
            </div>
            <div className="form-group">
              <label forHtml="encodeOutput" className="col-sm-2 control-label">Encoded</label>
              <div className="col-sm-10">
                <textarea className="form-control" id="encodeOutput"
                          value={this.state.encodeOutput} readOnly></textarea>
              </div>
            </div>
            <div className="form-group ">
              <div className="col-sm-offset-2 col-sm-10">
                <button id="encode" className="btn btn-primary btn-sm"
                        onClick={this.goEncode} type="button">Encode
                </button>
                <button className="btn btn-danger btn-sm"
                        onClick={this.clearEncode} type="button">Clear
                </button>
              </div>
            </div>
          </form>
        </tab>
        <tab heading="Decode" id="decodeTab">
          <form className="form-horizontal" role="form">
            <h2>Decode Base 64 to Data</h2>
            <div className="form-group">
              <label forHtml="decodeInput" className="col-sm-2 control-label">Encoded</label>
              <div className="col-sm-10">
                <textarea className="form-control" id="decodeInput"
                          value={this.state.decodeInput}
                          onChange={this.setDecodeInput} required></textarea>
              </div>
            </div>
            <div className="form-group">
              <label forHtml="decodeOutput" className="col-sm-2 control-label">Un-encoded</label>
              <div className="col-sm-10">
                <textarea className="form-control" id="decodeOutput"
                          value={this.state.decodeOutput} readOnly></textarea>
              </div>
            </div>
            <div className="form-group ">
              <div className="col-sm-offset-2 col-sm-10">
                <button id="decode" className="btn btn-primary btn-sm"
                        onClick={this.goDecode} type="button">Decode
                </button>
                <button className="btn btn-danger btn-sm"
                        onClick={this.clearDecode} type="button">Clear
                </button>
              </div>
            </div>
          </form>
        </tab>
      </tabset>
    </div>;
  }
});

export default Base64;
