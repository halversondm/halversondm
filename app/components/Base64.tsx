/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from "react";
import { type ReactNode, useState } from "react";

interface Base64State {
  encodeInput: string;
  encodeOutput: string;
  decodeInput: string;
  decodeOutput: string;
}

export default function Base64(): ReactNode {
  const [base64State, setBase64State] = useState<Base64State>({
    encodeInput: "",
    encodeOutput: "",
    decodeInput: "",
    decodeOutput: "",
  });

  function setEncodeInput(event): void {
    setBase64State({ ...base64State, encodeInput: event.target.value });
  }

  function goEncode(): void {
    const encodeOutput = window.btoa(base64State.encodeInput);
    setBase64State({ ...base64State, encodeOutput });
  }

  function clearEncode(): void {
    setBase64State({ ...base64State, encodeInput: "", encodeOutput: "" });
  }

  function setDecodeInput(event): void {
    setBase64State({ ...base64State, decodeInput: event.target.value });
  }

  function goDecode(): void {
    const decodeOutput = window.atob(base64State.decodeInput);
    setBase64State({ ...base64State, decodeOutput });
  }

  function clearDecode(): void {
    setBase64State({ ...base64State, decodeInput: "", decodeOutput: "" });
  }

  return (
    <div>
      <h4 className="text-success">
        An all Bootstrap and React implementation of converting your data to and
        from Base64 encoding. No tracking of your data is done. No calls to
        other services. It&quot;s all in your browser.
      </h4>
      <div>
        <div title="Encode" id="encodeTab">
          <form className="form-horizontal" role="form">
            <h2>Encode Data to Base64</h2>
            <div className="form-group">
              <label htmlFor="encodeInput" className="col-sm-2 control-label">
                Un-encoded
              </label>
              <div className="col-sm-10">
                <textarea
                  className="form-control"
                  id="encodeInput"
                  value={base64State.encodeInput}
                  onChange={setEncodeInput}
                  required={true}
                />
              </div>
            </div>
            <div className="form-group">
              <label htmlFor="encodeOutput" className="col-sm-2 control-label">
                Encoded
              </label>
              <div className="col-sm-10">
                <textarea
                  className="form-control"
                  id="encodeOutput"
                  value={base64State.encodeOutput}
                  readOnly={true}
                />
              </div>
            </div>
            <div className="form-group ">
              <div className="col-sm-offset-2 col-sm-10">
                <button
                  id="encode"
                  className="btn btn-primary btn-sm"
                  onClick={goEncode}
                  type="button"
                >
                  Encode
                </button>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={clearEncode}
                  type="button"
                >
                  Clear
                </button>
              </div>
            </div>
          </form>
        </div>
        <div title="Decode" id="decodeTab">
          <form className="form-horizontal" role="form">
            <h2>Decode Base 64 to Data</h2>
            <div className="form-group">
              <label htmlFor="decodeInput" className="col-sm-2 control-label">
                Encoded
              </label>
              <div className="col-sm-10">
                <textarea
                  className="form-control"
                  id="decodeInput"
                  value={base64State.decodeInput}
                  onChange={setDecodeInput}
                  required={true}
                />
              </div>
            </div>
            <div className="form-group">
              <label htmlFor="decodeOutput" className="col-sm-2 control-label">
                Un-encoded
              </label>
              <div className="col-sm-10">
                <textarea
                  className="form-control"
                  id="decodeOutput"
                  value={base64State.decodeOutput}
                  readOnly={true}
                />
              </div>
            </div>
            <div className="form-group ">
              <div className="col-sm-offset-2 col-sm-10">
                <button
                  id="decode"
                  className="btn btn-primary btn-sm"
                  onClick={goDecode}
                  type="button"
                >
                  Decode
                </button>
                <button
                  className="btn btn-danger btn-sm"
                  onClick={clearDecode}
                  type="button"
                >
                  Clear
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}
