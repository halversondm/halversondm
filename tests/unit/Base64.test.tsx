import React from "react";
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import Base64 from "../../app/components/Base64";

describe("Base64", () => {
    it("Will produce a snapshot of the Base64 component", () => {
        const component = render(<Base64 />);
        expect(component.container.toString()).toMatchSnapshot();
    });

    it("Will test the Base64 component encode functionality", async() => {
        render(<Base64 />);
        await userEvent.type(document.getElementById("encodeInput"), 'jpmc1234');
        await userEvent.click(document.getElementById("encode"));
        expect(document.getElementById("encodeOutput").value).toEqual('anBtYzEyMzQ=');
    });

    it("Will test the Base64 component decode functionality", async() => {
        render(<Base64 />);
        await userEvent.type(document.getElementById("decodeInput"), 'anBtYzEyMzQ=');
        await userEvent.click(document.getElementById("decode"));
        expect(document.getElementById("decodeOutput").value).toEqual('jpmc1234');
    });
})

