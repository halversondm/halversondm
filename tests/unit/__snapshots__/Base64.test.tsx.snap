// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base64 > Will produce a snapshot of the Base64 component 1`] = `"<div><div><h4 class=\\"text-success\\">An all Bootstrap and React implementation of converting your data to and from Base64 encoding. No tracking of your data is done. No calls to other services. It's all in your browser.</h4><div><div title=\\"Encode\\" id=\\"encodeTab\\"><form class=\\"form-horizontal\\" role=\\"form\\"><h2>Encode Data to Base64</h2><div class=\\"form-group\\"><label for=\\"encodeInput\\" class=\\"col-sm-2 control-label\\">Un-encoded</label><div class=\\"col-sm-10\\"><textarea class=\\"form-control\\" id=\\"encodeInput\\" required=\\"\\"></textarea></div></div><div class=\\"form-group\\"><label for=\\"encodeOutput\\" class=\\"col-sm-2 control-label\\">Encoded</label><div class=\\"col-sm-10\\"><textarea class=\\"form-control\\" id=\\"encodeOutput\\" readonly=\\"\\"></textarea></div></div><div class=\\"form-group \\"><div class=\\"col-sm-offset-2 col-sm-10\\"><button id=\\"encode\\" class=\\"btn btn-primary btn-sm\\" type=\\"button\\">Encode</button><button class=\\"btn btn-danger btn-sm\\" type=\\"button\\">Clear</button></div></div></form></div><div title=\\"Decode\\" id=\\"decodeTab\\"><form class=\\"form-horizontal\\" role=\\"form\\"><h2>Decode Base 64 to Data</h2><div class=\\"form-group\\"><label for=\\"decodeInput\\" class=\\"col-sm-2 control-label\\">Encoded</label><div class=\\"col-sm-10\\"><textarea class=\\"form-control\\" id=\\"decodeInput\\" required=\\"\\"></textarea></div></div><div class=\\"form-group\\"><label for=\\"decodeOutput\\" class=\\"col-sm-2 control-label\\">Un-encoded</label><div class=\\"col-sm-10\\"><textarea class=\\"form-control\\" id=\\"decodeOutput\\" readonly=\\"\\"></textarea></div></div><div class=\\"form-group \\"><div class=\\"col-sm-offset-2 col-sm-10\\"><button id=\\"decode\\" class=\\"btn btn-primary btn-sm\\" type=\\"button\\">Decode</button><button class=\\"btn btn-danger btn-sm\\" type=\\"button\\">Clear</button></div></div></form></div></div></div></div>"`;
